Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,255 @@
+        },
+
+        /**
+         * Assign the role to the users/groups after the user has chosen them in
+         * the universal discovery widget
+         *
+         * @method _assignRole
+         * @protected
+         * @param {EventFacade} e
+         */
+        _assignRole: function (e) {
+            var data = e.target.get('data'),
+                userService = this.get('capi').getUserService(),
+                service = this;
+
+            this._assignRoleNotificationStarted(
+                data.roleId, data.roleName, e.selection
+            );
+
+            userService.loadRole(data.roleId, function (error, response) {
+                var tasks = new Y.Parallel(),
+                    end = tasks.add(function (err, response) {
+                        if (err) {
+                            service._error('tasks.add() failed: ' + err.message);
+                            return;
+                        }
+                    });
+
+                if (error) {
+                    service._error('loadRole failed: ' + error.message);
+                    return;
+                }
+
+                service._assignRoleToEachAssignee(e, response.document.Role, end);
+
+                tasks.done(function () {
+                    service._assignRoleCallback(
+                        data.roleId, data.roleName, e.selection,
+                        Y.bind(data.afterUpdateCallback, service)
+                    );
+                });
+            });
+        },
+
+        /**
+         * Assign the role to each user/group
+         *
+         * @method _assignRoleToEachAssignee
+         * @protected
+         * @param {EventFacade} e
+         * @param {Role} role
+         * @param {Function} end
+         */
+        _assignRoleToEachAssignee: function (e, role, end) {
+            var userService = this.get('capi').getUserService(),
+                service = this,
+                roleAssignInputStruct;
+
+            Y.Array.each(e.selection, function (struct) {
+                var limitation = null,
+                    contentTypeIdentifier = struct.contentType.get('identifier');
+
+                // TODO: EZP-24905 Role assignment limitations
+                // limitation will be filled like example in UserService.prototype.newRoleAssignInputStruct()
+                // hrefs in the limitation must be converted to IDs, as expected by the API
+
+                roleAssignInputStruct = userService.newRoleAssignInputStruct(role, limitation);
+
+                if (contentTypeIdentifier === 'user') {
+                    service._assignRoleToUser(roleAssignInputStruct, struct.content, end);
+                } else if (contentTypeIdentifier === 'user_group') {
+                    service._assignRoleToUserGroup(roleAssignInputStruct, struct.content, end);
+                } else {
+                    service._error('The selected content "' + struct.content.get('name') + '" is not a user or user group.');
```

Comment:
There is no `end` callback called and I'm wondering if whole process won't fail? I guess that instead of triggering error notification you cald call `end(true);` which will trigger error in `end` function.